genie-web/src/main/java/com/netflix/genie/web/jpa/entities/v4/EntityDtoConverters.java

```diff
@@ -243,14 +242,23 @@ public static JobRequest toV4JobRequestDto(
         );
     }
 
-    private static Criterion toCriterionDto(final CriterionEntity criterionEntity) throws GeniePreconditionException {
```

Comment:
This class does not specify or know where the record is coming from. It's a generic converter and as such it can be fed bad inputs that can fail.
That's why I think it should explicitly declare an exception and force the using class to deal with it.

Then, in places where we use this class to convert a record *that we know and expect to be valid* we could stop propagating this exception.